keras_cv/metrics/object_detection/box_coco_metrics.py

```diff
@@ -0,0 +1,285 @@
+import os
+import sys
+import types
+
+import tensorflow as tf
+import tensorflow.keras as keras
+
+import keras_cv
+from keras_cv import bounding_box
+
+
+class HidePrints:
+    """A basic internal only context manager to hide print statements."""
+
+    def __enter__(self):
+        self._original_stdout = sys.stdout
+        sys.stdout = open(os.devnull, "w")
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        sys.stdout.close()
+        sys.stdout = self._original_stdout
+
+
+def _box_concat(b1, b2):
+    """Concatenates two bounding box batches together."""
+    if b1 is None:
+        return b2
+    if b2 is None:
+        return b1
+
+    result = {}
+    for key in ["boxes", "classes"]:
+        result[key] = tf.concat([b1[key], b2[key]], axis=0)
+
+    if "confidence" in b2:
+        result["confidence"] = tf.concat(
+            [b1["confidence"], b2["confidence"]], axis=0
+        )
+    return result
+
+
+METRIC_NAMES = [
+    "AP",
+    "AP50",
+    "AP75",
+    "APs",
+    "APm",
+    "APl",
+    "ARmax1",
+    "ARmax10",
+    "ARmax100",
+    "ARs",
+    "ARm",
+    "ARl",
+]
+
+
+class BoxCOCOMetrics(keras.metrics.Metric):
+    """BoxCOCOMetrics computes standard object deteciton metrics.
+
+    Args:
+        bounding_box_format: the bounding box format for inputs.
+        evaluate_freq: the number of steps to run before each evaluation.
+            Due to the high computational cost of metric evaluation the final
+            results are only updated once every `evaluate_freq` steps.  Higher
+            values will allow for faster training times, while lower numbers
+            allow for higher numerical precision in metric reporting.
+
+    Usage:
+    `BoxCOCOMetrics()` can be used like any standard metric with any
+    KerasCV object detection model.  Inputs to `y_true` must be KerasCV bounding
+    box dictionaries, `{"classes": classes, "boxes": boxes}`, and `y_pred` must
+    follow the same format with an additional `confidence` key.
+
+    Using this metric suite alongside a model is trivial; simply provide it to
+    the `compile()` arguments of the model:
+
+    ```python
+    images = tf.ones(shape=(1, 512, 512, 3))
+    labels = {
+        "boxes": [
+            [
+                [0, 0, 100, 100],
+                [100, 100, 200, 200],
+                [300, 300, 400, 400],
+            ]
+        ],
+        "classes": [[1, 1, 1]],
+    }
+    model = keras_cv.models.RetinaNet(
+        num_classes=20,
+        bounding_box_format="xywh",
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        classification_loss='focal',
+        box_loss='smoothl1',
+        optimizer=tf.optimizers.SGD(global_clipnorm=10.0),
+        jit_compile=False,
+        metrics=[keras_cv.metrics.BoxCOCOMetrics('xywh')]
+    )
+    model.fit(images, labels)
+    ```
+    """
+
+    def __init__(self, bounding_box_format, eval_steps, name=None, **kwargs):
```

Comment:
Update the arg name here and in the code please

Reply:
Totally missed this - thanks.